src/io/pti.jl

```diff
@@ -943,4 +943,744 @@ function _populate_defaults!(data::Dict)
+        if branch["transformer"]
+
+            if data["source_type"] == "matpower"
+                k = 0
+            else 
+                _, i, j, k, ckt, _ = branch["source_id"]
+            end
+
+            if k != 0 # Tree Winding
+                side = branch["f_bus"]
+            
+                if side == i
+                    three_winding_tran[i, j, k, ckt]["w1"] = branch
+                elseif side == j
+                    three_winding_tran[i, j, k, ckt]["w2"] = branch
+                else
+                    three_winding_tran[i, j, k, ckt]["w3"] = branch
+                end
+
+                windings = [
+                    three_winding_tran[i, j, k, ckt]["w1"],
+                    three_winding_tran[i, j, k, ckt]["w2"],
+                    three_winding_tran[i, j, k, ckt]["w3"],
+                ]
+
+                # if 3w Transformer Dict is full 
+                if ! (nothing in windings)
+                    push!(transformers, ("3W", three_winding_tran[i, j, k, ckt]))
+                end
+
+            else # Two Winding    
+                push!(transformers, ("2W", branch))
+            end            
+
+        else
+            # Get default owner
+            bus_i = branch["f_bus"]
+            owner = bus_owner[bus_i]
+            
+            # Get Dict in a PSSE way
+            psse_comp = _pm2psse_branch(branch, owner)
+            
+            # Print it in the file
+            _print_pti_str(io, psse_comp, _pti_dtypes["BRANCH"])
+        end
+    end
+    
+    println(io, "0 / END OF BRANCH DATA, BEGIN TRANSFORMER DATA")
+    
+    # Transformers
+    for (type, transformer) in transformers
+        # Get default transformer base
+        sbase = data["baseMVA"]
+
+        if type == "2W"
+            # Get default owner
+            bus_i = transformer["f_bus"]
+            owner = bus_owner[bus_i]
+                        
+            # Get the source data
+            source_type = data["source_type"]
+
+            # Convert to two winding transformer
+            psse_comp = _pm2psse_2w_tran(transformer, owner, sbase, source_type)
+            
+            lines = [
+                "TRANSFORMER",
+                "TRANSFORMER TWO-WINDING LINE 1",
+                "TRANSFORMER TWO-WINDING LINE 2",
+                "TRANSFORMER TWO-WINDING LINE 3",
+                ]
+                
+        elseif type == "3W"
+            # Get default owner
+            bus_i = transformer["w1"]["f_bus"]
+            owner = bus_owner[bus_i]
+            
+            # Convert to Three winding transformer
+            psse_comp = _pm2psse_3w_tran(transformer, owner, sbase)
+            
+            lines = [
+                "TRANSFORMER",
+                "TRANSFORMER THREE-WINDING LINE 1",
+                "TRANSFORMER THREE-WINDING LINE 2",
+                "TRANSFORMER THREE-WINDING LINE 3",
+                "TRANSFORMER THREE-WINDING LINE 4",
+                ]
```

Comment:
As a precaution, this can have an else block with `@assert false # transformer type not defined.`

Reply:
Done according to the advice.